sootup.core/src/main/java/sootup/core/cache/LRUCache.java

```diff
@@ -0,0 +1,58 @@
+package sootup.core.cache;
+
+import java.util.*;
+import javax.annotation.Nonnull;
+import sootup.core.model.SootClass;
+import sootup.core.types.ClassType;
+
+/**
+ * Cache that implements a least recently used strategy. If the amount of stored classes exceeds a
+ * specified amount, the lest recently used class will be overwritten.
+ */
+public class LRUCache<S extends SootClass<?>> implements Cache<S> {
+  private final int cacheSize;
+  private final Map<ClassType, S> cache = new HashMap<>();
+  private final LinkedList<ClassType> accessOrder = new LinkedList<>();
+
+  public LRUCache(int cacheSize) {
+    this.cacheSize = cacheSize;
+  }
+
+  @Override
+  public synchronized S getClass(ClassType classType) {
+    S sootClass = cache.get(classType);
+    if (sootClass != null) {
+      accessOrder.remove(classType);
+      accessOrder.addFirst(classType);
+    }
+
```

Comment:
should we throw an exception?

Reply:
I think we should stick to default Map behavior and return null, if the element is not present.